src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -414,10 +416,21 @@ MutationsInterpreter::MutationsInterpreter(
     ContextPtr context_,
     Settings settings_)
     : MutationsInterpreter(
-        Source(storage_, std::move(source_part_), std::move(alter_conversions_)),
+        Source(storage_, source_part_, std::move(alter_conversions_)),
         std::move(metadata_snapshot_), std::move(commands_),
         std::move(available_columns_), std::move(context_), std::move(settings_))
 {
+    const auto & part_columns = source_part_->getColumnsDescription();
```

Comment:
perhaps we should move the changes to this constructor https://github.com/ClickHouse/ClickHouse/blob/91d615812e809953142ad2854c768181e4a7b51c/src/Interpreters/MutationsInterpreter.cpp#L436
as its called by all others, what do you think ?

Reply:
Looks like we are doing similar things in `getAvailableColumnsWithVirtuals`, maybe we can combine and use either this function or move everything to common constructor

Reply:
This constructor with data part has a bit different logic. I'm afraid that I cannot unify them.